### PR TITLE
Merge Franz 5.5.0 beta.1 and 2

### LINKED ISSUE #280: Allow to disable hibernation for individual services
One thing I've noticed, which is really troublesome for me in comparison to Rambox, is that services seem to be shut down (presumably to save memory) after a certain amount of time. I'm not sure what this timeout limit is, but it would be good if we could define this limit to even include "never". I have some services that simply need to be open always in order to be able to respond more quickly to some customers vs having the service load back up.

### LINKED ISSUE #551: Specify a unique User-Agent
It appears that Ferdi sends the following User-Agent to servers:

```
User agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.130 Safari/537.36
```

It'd be nice if there were something here that indicated that the connection is from Ferdi, not from Chrome, ideally containing the Ferdi version.  (E.g. due to #547, Ferdi/Franz recipes for Zulip throw exceptions every time a new message arrives in Zulip, and it'd be nice to be able to filter those)

### PR DESCRIPTION
**Pull request by vantezzen**
### Description
This PR will merge Franz 5.5.0-beta.1 and the current progress of beta.2 into Ferdi. This is mainly to merge the fix for #264 and other fixes for current issues in Franz and Ferdi.

This PR also closes #280 as it adds a new "Disable hibernation" option for each service and closes #551 as it will add more information about Ferdi to the user agent.

### How Has This Been Tested?

This has been tested by trying different features of Ferdi. Everything seems to work but we still need to further test if everything works (especially login/registration and using without an account).

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project (run `$ yarn lint`).

### CODE REVIEW

**Comment by vantezzen** [comment]
Did some testing on our Windows VM and everything seems to work fine there

**Comment by kytwb** [comment]
I can't seem to run Ferdi locally following the last commit:

Screen Shot 2020-04-09 at 3 44 48 PM

**Comment by kytwb** [comment]
I'm getting the following when trying to review locally:

Screen Shot 2020-04-09 at 3 47 17 PM

**Comment by vantezzen** [comment]
Hmm, have you rerun `npx lerna bootstrap`? Some dependencies have changed (but I think Electron stayed the same - just to make sure).

**Comment by kytwb** [comment]
@vantezzen I did, but was a bit in a hurry too, couldn't take a longer look. Taking another look now.

**Comment by kytwb** [comment]
Still facing the same issue; removed the `build`/`node_modules` folder, fresh `npx lerna bootstrap` then `npm run dev` and `npm run start` (or `npx misty`). Will try again later.

**Comment by kytwb** [comment]
@vantezzen It's only with this file too; I mocked the undefined variable and it works. I tested a bit too; noticed WhatsApp shows the Google Chrome update; refreshing the service doesn't fix it. Otherwise Gmail works.

Screen Shot 2020-04-10 at 10 29 33 PM

**Comment by kytwb** [comment]
I will continue the review more in-depth later, noticed a couple of things already regarding the travis config, the macOS entitlements, need to test screen sharing etc.

**Comment by kytwb** [comment]
Quick update, will have to get back to it later 🙏 

Screen Shot 2020-04-11 at 7 04 36 PM

Screen Shot 2020-04-11 at 7 04 30 PM

**Comment by vantezzen** [comment]
@kytwb Any updates from your side? I can't reproduce your errors on my device nor on the Windows VM - it looks like your Electron dependency doesn't get correctly linked in the files that changed but I don't know why that would happen.

**Comment by kytwb** [comment]
I managed to review other pull requests locally, but couldn't get this one to run. I am not sure why yet. I should be able to take another look at it after working hours here.

**Comment by kytwb** [comment]
I feel a bit stupid to be blocking this pull request 😅 @vantezzen if you are confident, feel free to proceed, I don't want to be blocking more 🙏 Otherwise please bear with me until I have more time to address / review it. Starting a quick review without running locally in the meantime.

File: .travis.yml [hunk 1]
```diff
@@ -26,7 +26,9 @@ before_script:
   - travis_retry npx lerna bootstrap
   - travis_retry npm install node-sass -g
 script:
-  - travis_retry travis_wait 100 npm run build
+  - npm run test
+  - travis_wait 45 npm run build
```
Comment (kytwb) [inline .travis.yml:30]: I think these changes can be reverted?

File: build-helpers/default.entitlements.mac.plist [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kytwb) [inline build-helpers/default.entitlements.mac.plist:1]: Not needed as far as I know (we use entitlements.mas.plist and entitlements.mas.inherit.plist).

File: package.json [hunk 3]
```diff
@@ -85,8 +86,10 @@
     "gulp-uglify": "3.0.2",
     "hex-to-rgba": "1.0.2",
     "jsonwebtoken": "8.5.1",
-    "lodash": "4.17.15",
-    "marked": "0.7.0",
+    "lodash": "^4.17.4",
+    "mac-screen-capture-permissions": "1.1.0",
```
Comment (kytwb) [inline package.json:90]: I am not sure if we need a special entitlement for macOS for screen capture/sharing to work on macOS; that would need to be tested with a signed/notarized binary.

**Comment by kytwb** [comment]
I am blocked and can't review this pull request further. I have asked for help to @vantezzen in private to eventually arrange a pairing session so I can share my screen and pick his eyes/brain in case I'm missing something 🙏 Thanks for your patience.

**Comment by AlexOhayon** [comment]
Do you have an ETA of when this fix will be released to the next beta ? I'll be very happy to test it in beta ! Thanks

**Comment by vantezzen** [comment]
@AlexOhayon No ETA yet but it shouldn't take too long. I'd like kytwb or other contributors to test it before we merged it but after that we will release 5.4.4-gm pretty quickly

**Comment by AlexOhayon** [comment]
great, thanks for the update !

**Comment by vantezzen** [comment]
@kytwb  Looks like we might have a problem in the `ui` package. That might also cause your issue?

```
src/badge/ProBadge.tsx(2,23): error TS2307: Cannot find module '@meetfranz/theme'.
src/badge/index.tsx(1,23): error TS2307: Cannot find module '@meetfranz/theme'.
src/headline/index.tsx(1,23): error TS2307: Cannot find module '@meetfranz/theme'.
src/icon/index.tsx(2,23): error TS2307: Cannot find module '@meetfranz/theme'.
src/infobox/index.tsx(2,23): error TS2307: Cannot find module '@meetfranz/theme'.
src/loader/index.tsx(1,23): error TS2307: Cannot find module '@meetfranz/theme'.
src/typings/generic.ts(1,23): error TS2307: Cannot find module '@meetfranz/theme/lib'.
lerna ERR! npm run prepare stderr:
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @meetfranz/ui@1.0.0 prepare: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @meetfranz/ui@1.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by kytwb** [comment]
The situation here seem to be unlocked 🎉🙏 Thank you @vantezzen! 
I will review and merge today!

**Comment by kytwb** [comment]
We still need to double check the signed/notarized binary carefully before publishing the next release.

**Pull request merged**